docs/workflows/common-workflow-steps/custom-script.md

```diff
@@ -227,87 +227,139 @@ sudo -E printenv
-then
-	os="darwin"
-elif uname -a | grep -iq "linux"
-then
-	os="linux"
-fi
+# Set SMTP server 
+HOST_="smtp.gmail.com"
+PORT_="587"
+ACCOUNT_="gmail"
 
-if [ "$os" == "" ]
-then
-	echo "This script expects darwin or linux."
-	exit 1
-elif [ "$os" == "darwin" ] && which brew | grep -c "not found"
-then
-	echo "Can't find brew installation; make brew command visible or install homebrew and try again"
-	exit 1
-elif [ "$os" == "linux" ] && ! dpkg -s apt | grep -c "install ok installed"
-then
-	echo "apt is not installed; install apt and try again"
-	exit 1
-fi
+# Add account details (make sure to replace EMAIL_, USERNAME_, and PASSWORD_ with your actual details)
+EMAIL_="your-email-address@gmail.com"
+USERNAME_="your-email-address@gmail.com"
+PASSWORD_="your-email-password"
 
-if [ "$os" == "linux" ]
-then
-	sudo apt-get install mailutils
-	sudo apt-get install msmtp
-	sudo apt-get install msmtp-mta
-fi
+# Set e-mail details
+EMAIL_SUBJECT="Test Email Subject"
+EMAIL_TO="recieve-email-address"
+# This part will be used for visualization, Example usage: "Appcircle <test@gmail.com>"
+EMAIL_FROM="sender-email-address"
+EMAIL_BODY="This is the body of the test email."
 
-if [ "$os" == "darwin" ]
-then
-	brew install mailutils
-	brew install msmtp
-	echo "set sendmail=/usr/local/bin/msmtp" | sudo tee -a /etc/mail.rc
+# Detect operating system
+os=""
+if uname -a | grep -iq "darwin"; then
+    os="darwin"
+elif uname -a | grep -iq "linux"; then
+    os="linux"
 fi
 
-echo "defaults
-auth on
-tls on" >> ~/.msmtprc
-
-if [ "$os" == "linux" ]
-then
-	echo "tls_trust_file /etc/ssl/certs/ca-certificates.crt" >> ~/.msmtprc
-elif [ "$os" == "darwin" ]
-then
-	{ echo -n "tls_fingerprint " &
-	  msmtp --serverinfo --tls --tls-certcheck=off --host=smtp.gmail.com --port=587 \
-	  | egrep -o "([0-9A-Za-z]{2}:){31}[0-9A-Za-z]{2}" ;
-	} >> ~/.msmtprc
+# Check if OS is supported and install necessary packages
+if [ "$os" == "darwin" ]; then
+    if ! command -v brew > /dev/null 2>&1; then
+        echo "Can't find brew installation; make brew command visible or install homebrew and try again"
+        exit 1
+    fi
+    brew install mailutils
+    brew install msmtp
+    echo "set sendmail=/usr/local/bin/msmtp" | sudo tee -a /etc/mail.rc
+    { echo -n "tls_fingerprint " && msmtp --serverinfo --tls --tls-certcheck=off --host=$HOST_ --port=$PORT_ | egrep -o "([0-9A-Za-z]{2}:){31}[0-9A-Za-z]{2}"; } >> ~/.msmtprc
+
+elif [ "$os" == "linux" ]; then
+    if ! command -v apt > /dev/null 2>&1; then
+        echo "apt is not installed; install apt and try again"
+        exit 1
+    fi
+    apt-get update
+    apt-get install -y mailutils msmtp msmtp-mta
+    echo "tls_trust_file /etc/ssl/certs/ca-certificates.crt" >> ~/.msmtprc
+else
+    echo "Unsupported OS: $os. This script expects Darwin or Linux."
+    exit 1
 fi
 
-echo "logfile ~/.msmtp.log
-account gmail
-host smtp.gmail.com
-port 587
+# Create the .msmtprc file with appropriate permissions
+cat <<EOF > ~/.msmtprc
```

Comment:
In 270 and 279, text is added to the `msmtprc` file, but since ">" is used in the command here, it recreates the content of the `msmtprc` file if it exists.

Therefore, this code should be before the if condition on line 261